Layout Editor refactoring

bobjacobsen opened the pull request:
(This is opened now to get CI feedback, as running `ant warnings` locally doesn't see to catch things for me; this note will be updated before the PR is made final)

 - Start move of View-specific LayoutEditor class members and methods to a separate LayoutEditorViewContext class.  This may eventually become the LayoutEditorView class (which may get a better name)

 - SignalMastLogic now writes XML form with various sub-elements in sorted order.  Previously, the order was based on the Object sort, so was essentially random.  This makes it easier to use comparison files.

- Added a Comparator class for NamedBeanHandles.  It's just a simple one based on the default bean sort via system names, which is sufficient for testing purposes.

- Improve how reference PNGs are written from LayoutEditor reference panels in layoutEditor/LoadAndStoreTest by making sure the block routing is done before continuing.  Also, limit the size of the .png

- add final, Javadoc in various places

- Update ArchUnit test files to clear current content.

- Update memory for appveyor Ant target

pabender commented:
@bobjacobsen the two panel files that failed to load in appveyor:

     [java] Failed scenarios:
     [java] java/acceptancetest/features/web/WebPanel.feature:23 # Web Panel requests
     [java] java/acceptancetest/features/web/WebPanel.feature:24 # Web Panel requests
     [java] 
     [java] 41 Scenarios (2 failed, 39 passed)

are:

https://github.com/JMRI/JMRI/blob/bd85548837050f1f46343ee01a0af65cceaefb38/java/acceptancetest/features/web/WebPanel.feature#L23-L24

It looks like those also failed on travis headed, so this appears to be a real error introduced by this PR.

pabender commented:
> It looks like those also failed on travis headed, so this appears to be a real error introduced by this PR.

Actually, the files were renamed without updating all the tests that use them.....

bobjacobsen commented:
> Actually, the files were renamed without updating all the tests that use them.....

Thanks, didn't realize they were used that far away.  Fixed, and added some comments to (hopefully) avoid recurrence.

pabender commented:
> > Actually, the files were renamed without updating all the tests that use them.....
> 
> Thanks, didn't realize they were used that far away. Fixed, and added some comments to (hopefully) avoid recurrence.

The reference to them there is checking the javascript code loads the panels OK.  It necessarily has to load the panel file first, then opens it with the web browser.

This is where #8467 becomes important to this whole project.
